docs/elements.md

```diff
@@ -1244,6 +1244,55 @@ As of right now, the element supports displaying either Pandas DataFrames as an
 - [`pl-code` to display blocks of code with syntax highlighting](#pl-code-element)
 - [`pl-variable-output` for displaying a matrix or element in code form.](#pl-variable-output-element)
 
+### `pl-dataframe` element
+
+Displays a formatted display of Pandas DataFrames, with various options for displaying types of columns and code for reproducing the DataFrame.
+
+#### Sample elements
+
+**question.html**
+
+```html
+<pl-dataframe params-name="df" show-index="false" show-dimensions="false" digits="4"></pl-dataframe>
+```
+
+**server.py**
+
+```python
+import prairielearn as pl
+import pandas as pd
+
+def generate(data):
+  df = pd.io.parsers.read_csv("breast-cancer-train.dat", header=None)
+  data["params"]["df"] = pl.to_json(df.head(15))
+```
+
+#### Customizations
+
+| Attribute         | Type                  | Default | Description                                                                                                                                                             |
+| ----------------- | --------------------- | ------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `params-name`     | string                | —       | The name of the key in `data['params']` to get a value from.                                                                                                            |
+| `show-header`     | boolean               | true    | Show the header row of a DataFrame.                                                                                                                                     |
+| `show-index`      | boolean               | true    | Show the index column of a DataFrame.                                                                                                                                   |
+| `show-dimensions` | boolean               | true    | Show a footer with the dimensions of a DataFrame.                                                                                                                       |
+| `show-dtype`      | "none", "python", "r" | "none"  | Show the data types contained in each column of the DataFrame at the bottom of each column corresponding to the language given, where "none" doesn't display the types. |
```

Comment:
@eliotwrobson I think we need to keep the `show-dtype` as a bool here for consistency with other options.

How about we introduce maybe `language` that stores `python` or `r` with the default being `python`?

Reply:
I think that makes sense, if the language will also affect the indexing. I'll add the parameter then.

Reply:
Excellent! I think that'll help out greatly.